src/app-layer-http2.c

```diff
@@ -0,0 +1,71 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Philippe Antoine <p.antoine@catenacyber.fr>
+ *
+ * Parser for HTTP2, RFC 7540
+ */
+
+#include "suricata-common.h"
+#include "stream.h"
+#include "conf.h"
+
+#include "util-unittest.h"
+
+#include "app-layer-detect-proto.h"
+#include "app-layer-parser.h"
+
+#include "app-layer-http2.h"
+#include "rust.h"
+
+static int HTTP2RegisterPatternsForProtocolDetection(void)
+{
+    //TODO is this too restrictive and can be evaded ?
+    /* TODO investigate why this makes AppLayerTest09 fail/leak
+     * and not leak if we only give a 5-bytes pattern
```

Comment:
Any thoughts here on the length on the pattern ?

Reply:
What is the max size you can use before other things break?

Reply:
16 bytes pattern "PRI * HTTP/2.0\r\n" works
18 bytes pattern  "PRI * HTTP/2.0\r\n\r\n" does not

Reply:
Ok lets go for the 16 byte pattern for now. Maybe add a note to a relevant ticket, or add a new ticket, for the length issue.

Reply:
Ok